### PR TITLE
Improved map clickspots

### PR DESCRIPTION
**Pull request by FX5F**
### 🔗 Linked Issue

#163, #149 

- [ ] 🐞 Bug fix
- [x] 👌 Enhancement (improve something existing)
- [ ] ✨ New functionality
- [ ] 🧹 Technical change (refactoring, updates and other improvements)

### 📚 Description

- Pilot popups can now be closed by clicking on the aircraft icon/label
- Give airport label click priority over aircraft icons
- Selecting a pilot at the airport dashboard now selects the airplane at the map and when selecting an aircraft at the map it now opens the pilot overlay at the airport dashboard
- Not unselecting the hovered aircraft at the manualHide event. reason: We do not set  hoveredAircraft = false in the manualHide event, because this led to a short time when moving with the mouse from the label to the icon where the aircraft got a "not hovered" state. We just switch to ManualHover=false and let the pointermove function handle the removal of the hover state
- we don't handle the click for the label at the label anymore, we handle that with the handleClick function

### CODE REVIEW

File: src/components/map/aircraft/MapAircraftList.vue [hunk 1]
```diff
@@ -36,11 +37,32 @@ const map = inject<ShallowRef<Map | null>>('map')!;
...
+    if (event.data && 'selectedPilot' in event.data) mapStore.addPilotOverlay(event.data.selectedPilot.toString());
+}
+
+function sendSelectedPilotToDashboard(cid: number | null = null) {
+    const message = { selectedPilot: cid };
+    const targetOrigin = config.public.DOMAIN;
+    window.parent.postMessage(message, targetOrigin);
+}
+
+onMounted(() => {
```
Comment (daniluk4000) [inline src/components/map/aircraft/MapAircraftList.vue:61]: Add onBeforeUnmount to remove this event listener please

File: src/components/map/aircraft/MapAircraftList.vue [hunk 2]
```diff
@@ -36,11 +37,32 @@ const map = inject<ShallowRef<Map | null>>('map')!;
...
+    if (event.origin !== config.public.DOMAIN) {
+        return;
+    }
+    if (event.data && 'selectedPilot' in event.data) mapStore.addPilotOverlay(event.data.selectedPilot.toString());
+}
+
+function sendSelectedPilotToDashboard(cid: number | null = null) {
+    const message = { selectedPilot: cid };
+    const targetOrigin = config.public.DOMAIN;
+    window.parent.postMessage(message, targetOrigin);
```
Comment (daniluk4000) [inline src/components/map/aircraft/MapAircraftList.vue:58]: Can parent be undefined though?

File: src/pages/airport/[icao].vue [hunk 3]
```diff
@@ -263,6 +265,42 @@ useHead({
...
+    }
+    if (event.data && 'selectedPilot' in event.data) {
+        if (selectedPilot.value !== event.data.selectedPilot) {
+            selectedPilot.value = event.data.selectedPilot;
+            skipSelectedPilotWatch = true; // the change in the line above will trigger the watch of selectedPilot. But here in this function we have received the change from the iframe, so we need to skip the watch, because we don't need to send a message back to the iframe
+        }
+    }
+}
+
+onMounted(() => {
```
Comment (daniluk4000) [inline src/pages/airport/[icao].vue:284]: Also onBeforeUnmount there please

File: src/pages/airport/[icao].vue [hunk 4]
```diff
@@ -263,6 +265,42 @@ useHead({
...
+watch(selectedPilot, () => {
+    if (skipSelectedPilotWatch) {
+        skipSelectedPilotWatch = false;
+        return;
+    }
+    if (aircraft.value?.prefiles.find(x => x.cid === selectedPilot.value)) return; // we can not show prefiles on the map, because they are not connected
+
+    if (airportMapFrame.value && selectedPilot.value) {
+        const iframeWindow = airportMapFrame.value.contentWindow;
+        const message = { data: 'Your parent here', selectedPilot: selectedPilot.value };
```
Comment (daniluk4000) [inline src/pages/airport/[icao].vue:297]: Placeholder from stackoverflow in data or do we need this?

File: src/pages/airport/[icao].vue [hunk 5]
```diff
@@ -649,6 +687,8 @@ await setupDataFetch({
         }
 
         &_pilot {
+            position: absolute;
```
Comment (daniluk4000) [inline src/pages/airport/[icao].vue:690]: This worked fine before, you sure?

**Review by daniluk4000 — CHANGES_REQUESTED** [review]

File: src/pages/airport/[icao].vue (hunk 5 above)
Comment (FX5F) [reply src/pages/airport/[icao].vue:694]: Before the pilot overlay made the map smaller when the overlay opened, with this the overlay is above the map and does not change the map size. I thought it looks better when the map size is not changed by the overlay. But it is an optional change

**Pull request merged**